test/e2e/gateway/gateway_test.go

```diff
@@ -19,134 +19,150 @@ import (
 	gatewayv1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
 )
 
-type testGroup struct {
-	name    string
-	gateway *gatewayv1alpha1.Gateway
-	tests   map[string]func(t *testing.T, f *e2e.Framework)
+func TestGatewayAPI(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Gateway API tests")
 }
 
-// getTests defines the test groups that make up the Gateway test suite.
-// Each group shares a single Gateway definition.
-func getTests() []*testGroup {
-	var testGroups []*testGroup
-
-	testGroups = append(testGroups, &testGroup{
-		name: "default gateway",
-		tests: map[string]func(t *testing.T, f *e2e.Framework){
-			"001-path-condition-match":               testGatewayPathConditionMatch,
-			"002-header-condition-match":             testGatewayHeaderConditionMatch,
-			"003-invalid-forward-to":                 testInvalidForwardTo,
-			"005-request-header-modifier-forward-to": testRequestHeaderModifierForwardTo,
-			"005-request-header-modifier-rule":       testRequestHeaderModifierRule,
-		},
-		gateway: &gatewayv1alpha1.Gateway{
-			// Namespace and name need to match what's
-			// configured in the Contour config file.
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: "projectcontour",
-				Name:      "contour",
-			},
-			Spec: gatewayv1alpha1.GatewaySpec{
-				GatewayClassName: "contour-class",
-				Listeners: []gatewayv1alpha1.Listener{
-					{
-						Protocol: gatewayv1alpha1.HTTPProtocolType,
-						Port:     gatewayv1alpha1.PortNumber(80),
-						Routes: gatewayv1alpha1.RouteBindingSelector{
-							Kind: "HTTPRoute",
-							Namespaces: &gatewayv1alpha1.RouteNamespaces{
-								From: routeSelectTypePtr(gatewayv1alpha1.RouteSelectAll),
-							},
-							Selector: &metav1.LabelSelector{
-								MatchLabels: map[string]string{"app": "filter"},
+var _ = Describe("Gateway API", func() {
+	var f *e2e.Framework
+
+	BeforeEach(func() {
+		f = e2e.NewFramework(GinkgoT())
+	})
+
+	Describe("Insecure (Non-TLS) Gateway", func() {
+		var gateway *gatewayv1alpha1.Gateway
+
+		BeforeEach(func() {
+			gateway = &gatewayv1alpha1.Gateway{
+				// Namespace and name need to match what's
+				// configured in the Contour config file.
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "projectcontour",
+					Name:      "contour",
+				},
+				Spec: gatewayv1alpha1.GatewaySpec{
+					GatewayClassName: "contour-class",
+					Listeners: []gatewayv1alpha1.Listener{
+						{
+							Protocol: gatewayv1alpha1.HTTPProtocolType,
+							Port:     gatewayv1alpha1.PortNumber(80),
+							Routes: gatewayv1alpha1.RouteBindingSelector{
+								Kind: "HTTPRoute",
+								Namespaces: &gatewayv1alpha1.RouteNamespaces{
+									From: routeSelectTypePtr(gatewayv1alpha1.RouteSelectAll),
+								},
+								Selector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{"app": "filter"},
+								},
 							},
 						},
 					},
 				},
-			},
-		},
+			}
+
+			Expect(f.Client.Create(context.TODO(), gateway)).To(Succeed())
```

Comment:
if we're not sold on using ginkgo yet, I think we could still use testify for these, just maybe pass the `testing.T` argument as `GinkgoT()`?